### PR TITLE
Redirect to login page when logged out

### PR DESCRIPTION
**Pull request by arikmaor**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -20,6 +31,21 @@ const App: FC = () => {
   );
 };
 
+function WelcomePage() {
+  const { user, logout } = useAuth();
+  return (
+    <div>
+      Welcome {user?.local.name}
```
Comment (AsafYehezkelTL) [inline src/App.tsx:38]: if there is no user the perhaps the message should be changed to something else, WDYT?

File: src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
       <BrowserRouter>
         <Switch>
+          <Route exact path="/" component={RootRedirect} />
           <Route exact path="/login">
             <LoginPage />
           </Route>
+          <PrivateRoute exact path="/welcome">
+            <WelcomePage />
+          </PrivateRoute>
           <Route path="*">
```
Comment (AsafYehezkelTL) [inline src/App.tsx:25]: Consider writing the all `Route`s  with component attribute rather than component child

File: src/App.tsx [hunk 3]
```diff
@@ -32,4 +58,27 @@ function NoMatch() {
...
+        if (!user) {
+          return (
+            <Redirect
+              to={{
+                pathname: '/login',
+                state: { from: props.location },
+              }}
+            />
+          );
+        }
```
Comment (AsafYehezkelTL) [inline src/App.tsx:76]: consider declaring render function above like:
```typescript
const render = (props) => {
    if (!user) {
          return (
            <Redirect
              to={{
                pathname: '/login',
                state: { from: props.location },
              }}
            />
          );
```

File: src/core/local-storage.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        value instanceof Function ? value(storedValue) : value;
+      setStoredValue(valueToStore);
+      window.localStorage.setItem(key, JSON.stringify(valueToStore));
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  return [storedValue, setValue];
+}
```
Comment (AsafYehezkelTL) [inline src/core/local-storage.ts:30]: Cool!

**Review by AsafYehezkelTL — APPROVED** [review]
Looks awesome
Added few suggestions as non-blocking

File: src/App.tsx (hunk 1 above)
Comment (arikmaor) [reply src/App.tsx:38]: It's just a placeholder so we'll have something to show while logged on

Currently I've implemented the `PrivateRoute` component which is based [react-router documentation](https://reactrouter.com/web/example/auth-workflow) but we can also go create [a hook](https://usehooks.com/useRequireAuth/) which may be better as we can return a non-nullable user

File: src/App.tsx (hunk 2 above)
Comment (arikmaor) [reply src/App.tsx]: done

Comment (AsafYehezkelTL) [reply src/App.tsx]: Thanks

File: src/App.tsx (hunk 3 above)
Comment (arikmaor) [reply src/App.tsx:70]: ack

**Pull request merged**